Allow for _FillValue in time dimension

JoostBuitink opened the pull request:
Fixed an issue where the `_FillValue` attribute in the time dimension of the forcing file causes an error, as the `get_at` function was unable to dispatch due to unexpected types. This addition checks (if the `_FillValue`  is present) whether there are actual missing values in the time dimension. If this is the case, an error is thrown, otherwise the dates are converted to the correct type for the `get_at` function.

visr reviewed (approved):
Looks good to me! I assume you tested this locally. Might be good to add that is is not allowed according to CF. But this is nicer/more forgiving to users. If missings in practice often occur (e.g. from Delft-FEWS) we could probably even support it as discussed in https://github.com/Deltares/Wflow.jl/issues/221#issuecomment-1270695767, though if needed that can always be done later.

JoostBuitink commented:
Thanks for the quick review! I indeed tested this locally. I have added a warning message when `_FillValue` is present in the time dimension, to give feedback to the user. Also moved the check for actual missing values to outside of the `update_forcing!` function.
